File: solution/27065/Main.java
import java.io.*;

public class Main {
    static BufferedReader stdin = new BufferedReader(new InputStreamReader(System.in), 1048576);
    static StringBuilder stdout = new StringBuilder();
    public static void main(String[] args) throws Exception {
        int T = Integer.parseInt(stdin.readLine());
        for (int i = 0; i < T; ++i) {
            int n = Integer.parseInt(stdin.readLine());
            boolean ok = Main.isOver(n);
            for (int j = 2; j * j <= n; ++j) {
                if (n % j == 0) {
                    if (Main.isOver(j) || Main.isOver(n / j)) {
                        ok = false;
                        break;
                    }
                }
            }
            if (ok) {
                stdout.append("Good Bye\n");
            } else {
                stdout.append("BOJ 2022\n");
            }
        }
        System.out.println(stdout);
    }
    private static boolean isOver(int n) {
        int sum = 1;
        for (int i = 2; i * i <= n; ++i) {
            if (n % i == 0) {
                sum += i;
                int j = n / i;
                if (i != j) {
                    sum += j;
                }
            }
        }
        return sum > n;
    }
}

File: solution/27065/main.py
def is_over(n):
    s = 1
    i = 2
    while i * i <= n:
        if n % i == 0:
            s += i
            j = n // i
            if i != j:
                s += j
        i += 1
    return s > n


def main():
    input = open(0, "rb").readline
    T = int(input())
    for _ in range(T):
        n = int(input())
        ok = is_over(n)
        i = 2
        while i * i <= n:
            if n % i == 0:
                if is_over(i) or is_over(n // i):
                    ok = False
                    break
            i += 1
        if ok:
            print("Good Bye")
        else:
            print("BOJ 2022")


if __name__ == "__main__":
    main()
